src/System.Management.Automation/engine/COM/ComAdapter.cs

```diff
@@ -84,6 +84,45 @@ protected override T GetMember<T>(object obj, string memberName)
             return null;
         }
 
+        private protected override T GetMember<T>(object obj, MemberNamePredicate predicate)
+        {
+            bool lookingForProperties = typeof(T).IsAssignableFrom(typeof(PSProperty));
+            bool lookingForParameterizedProperties = typeof(T).IsAssignableFrom(typeof(PSParameterizedProperty));
+            if (lookingForProperties || lookingForParameterizedProperties)
+            {
+                foreach (ComProperty prop in _comTypeInfo.Properties.Values)
+                {
+                    if (prop.IsParameterized
+                        && lookingForParameterizedProperties
+                        && predicate.Invoke(prop.Name))
+                    {
+                        return new PSParameterizedProperty(prop.Name, this, obj, prop) as T;
+                    }
+
+                    if (lookingForProperties && predicate.Invoke(prop.Name))
```

Comment:
Seems it's more frequent and could be before previous "if".

Reply:
What I've done here is to follow the pattern in the rest of the file. If you look at the context, you will see that there are a lot of the same patterns. That doesn't come across when you only see the diff.

I can change this, but in that case, maybe it should be changed in the whole file?

Reply:
Oh, not in the PR - it is already huge. (Could we split the PR on some parts? Maybe we could move style/renaming changes in other PR and fast merge before the PR?)